src/main/java/com/cs544/project/controller/StudentViewController.java

```diff
@@ -0,0 +1,29 @@
+package com.cs544.project.controller;
+
+import com.cs544.project.domain.CourseOffering;
+import com.cs544.project.exception.CustomNotFoundException;
+import com.cs544.project.service.CourseOfferingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.crossstore.ChangeSetPersister;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.FileNotFoundException;
+import java.util.Collection;
+
+@RestController
+@RequestMapping("/student-view")
+public class StudentViewController {
```

Comment:
@bishosilwal chose to have resource-based controllers i.e. `CourseOfferingController` in this case, and have methods for student-view and I assume different views in the future i.e. `admin-view`, `sys-admin`, etc.

You're choosing view-based controllers.

Both options are okay, they both have pros and cons. We need to pick one and conform to it.


According to Gemini:

#### Approach 1: Resource-Based Controllers (CourseOfferingController, StudentsController, etc.)

**Pros:**
- Scalability: Handles many resources more easily, as each resource is encapsulated in its own controller.
- Clear Responsibility: Each controller has a single responsibility—managing a specific resource.
- Loose Coupling: Provides better separation between resource logic and view-specific logic.
- Flexibility: Makes it easier to adapt to changes in resources or views.

**Cons:**
- Code Duplication: If you have similar view-specific operations across resources, you might need to repeat code in multiple controllers.
- More Controllers: It can lead to a larger number of controllers, which might require more effort to manage.


#### Approach 2: View-Based Controllers (StudentViewController, AdminViewController, etc.)

**Pros:**
- Organization: Groups related resources by view, making the code more logical and easier to manage.
- Separation of Concerns: Clearly separates view-specific logic (e.g., data transformation for different views) from core resource logic.
- Less Code Repetition: If you have many resources that share similar view-specific operations, you can potentially reuse code within a view controller.

**Cons:**
- Potentially Large Controllers: If you have many resources and views, your view controllers could become very large and complex.
- Tight Coupling: Creates a tighter coupling between view-specific logic and resource logic, which might make it harder to refactor later.
- Limited Scalability: As the number of resources or views grows, managing these controllers could become more challenging.

Reply:
@SaurabGhimire if we decide on which option to use, the PR looks good 👍🏿 
but needs unit tests, unless you plan on doing that in another PR